services/spar/src/Spar/Scim/User.hs

```diff
@@ -406,6 +403,23 @@ calculateVersion uidText usr = Scim.Weak (Text.pack (show h))
     h :: Digest SHA256
     h = hashlazy (Aeson.encode (Scim.WithId uidText usr))
 
+{-|
+Check that the UserRef is not taken; or that it's taken by the given user id.
+
+FUTUREWORK: This check is very slow as it scans the every user for the team and decodes JSON
+to get the externalId. This could be cleaned up eventually by adding an index.
```

Comment:
The `FUTUREWORK` is off a little, we actually don't do anything inefficient, we just lookup the `UserRef`.  I would write this instead:

```haskell
-- ASSUMPTION: every scim user has a 'SAML.UserRef', and the `SAML.NameID` in it corresponds
-- to a single `externalId`.
--
-- FUTUREWORK: The @hscim@ library checks that the handle is not taken before 'create' is even
-- called.  However, it does that in an inefficient manner by pulling the entire user base of
-- this scim peer and searching it for the externalId in memory.  We should remove the check
-- from @hscim@ and rely only on the test here.  Also, make sure @hscim@ handles `externalId`
-- conflicts the same way in the create case and the update case.
```